File: Python/Priority queue/Remove Stones to Minimize the Total.py
class Solution:
    def minStoneSum(self, piles: List[int], k: int) -> int:
        pq=[]
        for i in piles:
            heapq.heappush(pq,-i)
        stones=0
        while k>0:
            k-=1
            np = -heapq.heappop(pq)
            s=np -math.floor(np/2)
            heapq.heappush(pq,-s)
        sum=0
        for i in pq:
            sum+=-i
        return sum
        #return -sum(pq)
        

File: CPP/Priority queue/Remove Stones to Minimize the Total.cpp
class Solution {
public:
    int minStoneSum(vector<int>& piles, int k) {
        priority_queue<int>pq;
        for(auto i:piles){
            pq.push(i);
        }
        int stones=0;
        while(k>0){
            k--;
            int np=pq.top();
            pq.pop();
            int s=np-floor(np/2.0);
            pq.push(s);
        }
        int sum=0;
        while(!pq.empty()){
            sum+=pq.top();
            pq.pop();
        }
        return sum;
    }
};
